### DISCUSSION TITLE
v1.99.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#8102] [Announcements]
# v1.99.0

## New logo
The new logo is now rolled out on all platforms, including the web, mobile, and documentation pages. If you are a content creator, you can grab the logo vector and raster files in the repository's `design` folder.

![Social Preview](https://github.com/immich-app/immich/assets/27055614/75d88502-1eec-4131-9c9e-340ac71a0e0c)

## Highlights

Welcome to release `v1.99.0` of Immich. This release adds the new logo, fixes many bugs, and improves the application. Some of the highlights include:

- [WEB] Sort photos and videos to display direction in an album.
- On-disk cache on the mobile app to provide a better viewing experience.
- There is a new endpoint for OpenTelemetry metrics to monitor performance.
- Optimized queries for large libraries.
- Add upload progress and speed indicator.
- XMP sidecar file is now recognized with both `photo.ext.xmp` and `photo.xmp` extensions.
- Drag to select - Now press your thumb to drag and select in the selection mode.
- Improve user feedback in the mobile app's backup and album selection screen.
- Various improvements for the library scanning process.

### Asset's display direction in the album

We added an option to change the order of the assets in the album so you can view the assets from older to newest and vice versa.

https://github.com/immich-app/immich/assets/27055614/7ed828c3-9bd1-4caa-8152-a7dc66a15cc5

### Mobile asset caching

Improving cache for remote thumbnails and the images, they now stay on disk so that they can load faster and give a better experience while used offline. You should feel a faster loading experience across the app.

### Performance monitoring with OpenTelemetry integration

Immich provides a variety of performance metrics to allow for local monitoring and insights. This integration is primarily in the form of Prometheus metrics. However, exporting traces is also possible due to OpenTelemetry instrumentation. Please visit the [documentation](https://immich.app/docs/features/monitoring) to learn how to enable this feature

> [!NOTE]
> This opt-in feature allows you to monitor Immich's performance. This data isn't sent anywhere beyond what you've configured.

### File size progress and upload speed indicator

We added an indicator to show the file's size that has been uploaded and the upload speed under the progress bar in the backup screen.
 
https://github.com/immich-app/immich/assets/7256684/5f06adc6-bd7f-4d82-9750-ba8a14628821

### Drag to select

You can now drag to select in the mobile app when in selection mode; this will help make selecting in bulk easier.

https://github.com/immich-app/immich/assets/139912620/e3c401bb-1221-4e33-acf2-de165ad30b38

**And lastly, I want to express my gratitude to @immich-app/contributor who make this behemoth of a release happen 🎉**

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 3QVAb9dCHutquVejeNXitPqZX26Yg5kxb7

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### 🗄️ Server
* fix(server): face search results not always sorted by @mertalev in https://github.com/immich-app/immich/pull/7839
* fix(server): serve static directory only if it exists by @benmccann in https://github.com/immich-app/immich/pull/7857
* feat(server): Add OAuth auto-redirect enable/disable CLI commands by @ddshd in https://github.com/immich-app/immich/pull/7841
* feat(server): OpenTelemetry integration by @mertalev in https://github.com/immich-app/immich/pull/7356
* feat(server): YAML config file support by @bo0tzz in https://github.com/immich-app/immich/pull/7894
* feat(server): Update XMP sidecar search to look for both photo.ext.xmp and photo.xmp by @kokulshan in https://github.com/immich-app/immich/pull/7813
* feat(server): lower library scan memory usage by @mertalev in https://github.com/immich-app/immich/pull/7939
* perf(server): optimize `getByIds` query by @mertalev in https://github.com/immich-app/immich/pull/7918
* fix(server): queue library asset refresh in batches by @etnoy in https://github.com/immich-app/immich/pull/7914
* feat(server, web): album orders by @martabal in https://github.com/immich-app/immich/pull/7819
* feat(server): added support for SVG files by @declan8010 in https://github.com/immich-app/immich/pull/7960
* refactor: global validation pipe by @jrasm91 in https://github.com/immich-app/immich/pull/7985
* feat(web): add millisecond options to storage template settings by @sevtdy in https://github.com/immich-app/immich/pull/7942
* fix(server): prevent feedback loop during library scan by @mertalev in https://github.com/immich-app/immich/pull/7944
* feat(server): use nestjs events to validate config by @danieldietzler in https://github.com/immich-app/immich/pull/7986
* feat(server): memory includes partners assets on timeline by @alextran1502 in https://github.com/immich-app/immich/pull/7993
* refactor(server): library service by @jrasm91 in https://github.com/immich-app/immich/pull/8050
* chore(server): change `save` -> `update` in asset repository by @mertalev in https://github.com/immich-app/immich/pull/8055
* fix(server): "view all" for cities only showing 12 cities by @mertalev in https://github.com/immich-app/immich/pull/8035
* fix(server): use extension in originalFileName for libraries by @etnoy in https://github.com/immich-app/immich/pull/8083
* test: temporarily disable flaky audit e2e test until #7436 is fixed by @zackpollard in https://github.com/immich-app/immich/pull/8089
### 📱 Mobile
* fix(mobile): Fixes immersive mode not ending for memory lane by @martyfuhry in https://github.com/immich-app/immich/pull/7767
* fix(mobile): Fixes local thumbnail image provider key by @martyfuhry in https://github.com/immich-app/immich/pull/7766
* Fix trash permanent delete dialog by @NicholasFlamy in https://github.com/immich-app/immich/pull/7805
* refactor(mobile): app settings by @shenlong-tanwen in https://github.com/immich-app/immich/pull/7749
* feat(mobile): add labels to app bar buttons by @ben-basten in https://github.com/immich-app/immich/pull/7865
* chore(mobile): Removes analysis options for openapi directory by @martyfuhry in https://github.com/immich-app/immich/pull/7309
* feat(mobile): update logo by @alextran1502 in https://github.com/immich-app/immich/pull/7919
* fix(mobile): make elements scrollable to avoid overflow in landscale by @thielepaul in https://github.com/immich-app/immich/pull/7933
* feat(mobile): Adds file upload progress stats by @othyn in https://github.com/immich-app/immich/pull/7760
* feat(mobile): Remote thumbnails and images use an on-disk image cache by @martyfuhry in https://github.com/immich-app/immich/pull/7929
* Make mobile app a "media management app" to get rid of android prompt by @Aegeontis in https://github.com/immich-app/immich/pull/7851
* refactor(mobile): backup album selection by @alextran1502 in https://github.com/immich-app/immich/pull/8053
* fix(mobile): do not show hidden people by @waclaw66 in https://github.com/immich-app/immich/pull/8072
* feat(mobile): drag to select assets by @shenlong-tanwen in https://github.com/immich-app/immich/pull/8004
* chore(mobile): svg logo by @alextran1502 in https://github.com/immich-app/immich/pull/8074
* fix(mobile): missing "Add name" translation by @waclaw66 in https://github.com/immich-app/immich/pull/8087
### 🖥️ Web
* feat(web): empty external library placeholder by @jrasm91 in https://github.com/immich-app/immich/pull/7848
* feat(web): add preference for thumbnail playback on hover by @samholton in https://github.com/immich-app/immich/pull/7831
* fix(web): small fixes for empty placeholder by @michelheusschen in https://github.com/immich-app/immich/pull/7859
* fix: Hide play bar when there's only one image in memory by @declan8010 in https://github.com/immich-app/immich/pull/7764
* fix(web): asset upload progress by @michelheusschen in https://github.com/immich-app/immich/pull/7882
* fix(web): asset viewer navigation buttons by @michelheusschen in https://github.com/immich-app/immich/pull/7888
* refactor(web): sidebar menu links by @michelheusschen in https://github.com/immich-app/immich/pull/7891
* feat(web): Implemented device last seen date and time with user locale support by @declan8010 in https://github.com/immich-app/immich/pull/7863
* feat(web): dark color-scheme css for dark mode by @ZTzTopia in https://github.com/immich-app/immich/pull/7812
* fix(web): prevent combobox options from disappearing by @michelheusschen in https://github.com/immich-app/immich/pull/7733
* fix(web): correctly use button and link elements by @michelheusschen in https://github.com/immich-app/immich/pull/7907
* fix(web): Remove excess spacing on #asset-grid and search bar by @natedawg in https://github.com/immich-app/immich/pull/7910
* fix(web): Enhance Notification Handling for Duplicate Assets by @declan8010 in https://github.com/immich-app/immich/pull/7858
* fix(web): url state of nested accordions by @danieldietzler in https://github.com/immich-app/immich/pull/7928
* feat(web): update logo by @alextran1502 in https://github.com/immich-app/immich/pull/7798
* fix(web): FullScreenContainer logo by @alextran1502 in https://github.com/immich-app/immich/pull/7938
* chore(web): Sharing -> Partner Sharing by @aviv926 in https://github.com/immich-app/immich/pull/7952
* refactor(web): albums list (1) by @martabal in https://github.com/immich-app/immich/pull/7660
* feat(web): preload assets in photo-viewer by @samholton in https://github.com/immich-app/immich/pull/7920
* feat(web): randomize password on reest by @samholton in https://github.com/immich-app/immich/pull/7943
* fix(web): keyboard shortcut handling by @michelheusschen in https://github.com/immich-app/immich/pull/7946
* fix(web): improve focus and shortcuts by @michelheusschen in https://github.com/immich-app/immich/pull/7983
* fix(web): minor album card issues by @michelheusschen in https://github.com/immich-app/immich/pull/7975
* chore(web): remove unused AssetSelectionViewer by @michelheusschen in https://github.com/immich-app/immich/pull/7981
* perf(web): batch asset store changes by @michelheusschen in https://github.com/immich-app/immich/pull/7974
* fix(web): logo invisible on share page by @michelheusschen in https://github.com/immich-app/immich/pull/7990
* refactor(web): list navigation with keyboard by @michelheusschen in https://github.com/immich-app/immich/pull/7987
* fix(web): updating asset store after remove by @michelheusschen in https://github.com/immich-app/immich/pull/7999
* fix(web): repair page typo by @danieldietzler in https://github.com/immich-app/immich/pull/8051
* fix(web): load panorama in shared link by @alextran1502 in https://github.com/immich-app/immich/pull/8060
* feat(web): combobox accessibility improvements by @ben-basten in https://github.com/immich-app/immich/pull/8007
* fix(web): prevent drag-n-drop upload overlay from showing when not dragging files by @Ethan13310 in https://github.com/immich-app/immich/pull/8082
* chore(web): add asset store unit tests by @michelheusschen in https://github.com/immich-app/immich/pull/8077
* fix(web): show search page errors and use feature flag by @michelheusschen in https://github.com/immich-app/immich/pull/8088
* fix(web): missing margin on people page by @Ethan13310 in https://github.com/immich-app/immich/pull/8081
* fix(web): shift-select by @alextran1502 in https://github.com/immich-app/immich/pull/8093
* fix(web): shift-select again by @alextran1502 in https://github.com/immich-app/immich/pull/8098
* fix(web): Share button visible when viewing album has only shared link by @alextran1502 in https://github.com/immich-app/immich/pull/8100
### 🧠 Machine Learning
* fix(server): ml gunicorn listen on ipv4 and ipv6 by default by @growse in https://github.com/immich-app/immich/pull/7930
* chore(deps): update onnxruntime-openvino by @mertalev in https://github.com/immich-app/immich/pull/7854
* feat(ml): add cache_dir option to OpenVINO EP by @dvdblg in https://github.com/immich-app/immich/pull/8018
### ⚡ CLI
* chore(cli): clarify use of concurrency option by @mertalev in https://github.com/immich-app/immich/pull/7840
* chore(cli): Version 2.1.0 by @bo0tzz in https://github.com/immich-app/immich/pull/7893
* refactor(cli): simplify by @jrasm91 in https://github.com/immich-app/immich/pull/7962
### 📓 Documentation
* chore: update readme with new logo and add design materials by @alextran1502 in https://github.com/immich-app/immich/pull/7852
* fix: immich logo links by @jrasm91 in https://github.com/immich-app/immich/pull/7856
* docs: Add guide for album sync python script by @markeeisner in https://github.com/immich-app/immich/pull/7736
* Update external-library.md by @GrandNewbien in https://github.com/immich-app/immich/pull/7850
* docs: backup-and-restore.md: fix broken link by @SlavikCA in https://github.com/immich-app/immich/pull/7806
* docs: config-file.md update config to current (v1.98) state by @SlavikCA in https://github.com/immich-app/immich/pull/7808
* fix(blog/docs): Fix Milestone linking for "multi select" and "View Exif" fix(docs): Update "zoodyy" to "yfrey" by @declan8010 in https://github.com/immich-app/immich/pull/7926
* docs: Update backup script example by @mmomjian in https://github.com/immich-app/immich/pull/7734
* feat(doc): new logo and screenshots by @alextran1502 in https://github.com/immich-app/immich/pull/7961
* fix(doc): logo size on small screen by @alextran1502 in https://github.com/immich-app/immich/pull/7992
* feat(doc) Updated feature-panel.png with the new logo by @declan8010 in https://github.com/immich-app/immich/pull/7995
* chore(docs): update readme by @alextran1502 in https://github.com/immich-app/immich/pull/8021
* fix(docs): add microservices to IMMICH_CONFIG_FILE env var documentation by @tylerbrockett in https://github.com/immich-app/immich/pull/8017
* feat(docs): User management new options by @aviv926 in https://github.com/immich-app/immich/pull/8029
* fix(docs): update authelia OIDC link by @JW-CH in https://github.com/immich-app/immich/pull/8070
* fix(docs) minor security warning raised by Borg by @thariq-shanavas in https://github.com/immich-app/immich/pull/8075
* docs(server): add documentation for prometheus metrics by @mertalev in https://github.com/immich-app/immich/pull/8084
### 🔨 Maintenance
* chore(deps): update open-api by @renovate in https://github.com/immich-app/immich/pull/7867
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/7866
* chore(deps): update dependency typescript to v5.4.2 by @renovate in https://github.com/immich-app/immich/pull/7861
* chore(deps): update dependency @types/node to v20.11.25 by @renovate in https://github.com/immich-app/immich/pull/7860
* fix(deps): update docs by @renovate in https://github.com/immich-app/immich/pull/7870
* chore(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7869
* fix(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/7871
* chore(deps): update base-image to v20240312 (major) by @renovate in https://github.com/immich-app/immich/pull/7892
* chore(deps): bump softprops/action-gh-release from 1 to 2 by @dependabot in https://github.com/immich-app/immich/pull/7844
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7880
* chore(deps): update grafana/grafana docker tag to v10.4.0 by @renovate in https://github.com/immich-app/immich/pull/7875
* chore: publish sdk on release by @jrasm91 in https://github.com/immich-app/immich/pull/7895
* fix(deps): update web by @renovate in https://github.com/immich-app/immich/pull/7872
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7898
* chore(deps): bump docker/build-push-action from 5.2.0 to 5.3.0 by @dependabot in https://github.com/immich-app/immich/pull/7947
* chore(deps): bump docker/setup-buildx-action from 3.1.0 to 3.2.0 by @dependabot in https://github.com/immich-app/immich/pull/7948
* chore(ci): Clean up docker buildx workaround by @bo0tzz in https://github.com/immich-app/immich/pull/7949
* chore(cli): clean up files by @jrasm91 in https://github.com/immich-app/immich/pull/7955
* chore: remove unused dep by @jrasm91 in https://github.com/immich-app/immich/pull/7957
* chore(cli): use lockfile v3 by @jrasm91 in https://github.com/immich-app/immich/pull/7958
* refactor: library e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7969
* chore(server): introduce proper job status by @danieldietzler in https://github.com/immich-app/immich/pull/7932
* chore(deps): pin prom/prometheus docker tag to bc1794e by @renovate in https://github.com/immich-app/immich/pull/7874
* fix(deps): update exiftool by @renovate in https://github.com/immich-app/immich/pull/7879
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/7890
* chore(deps): update typescript-eslint monorepo to v7.2.0 by @renovate in https://github.com/immich-app/immich/pull/8008
* chore: Simplify install script by @bo0tzz in https://github.com/immich-app/immich/pull/8048
* chore(deps): update dependency @types/node to v20.11.27 by @renovate in https://github.com/immich-app/immich/pull/8012
* chore: add prometheus.yml to release artifact by @alextran1502 in https://github.com/immich-app/immich/pull/8096

## New Contributors
* @GrandNewbien made their first contribution in https://github.com/immich-app/immich/pull/7850
* @ZTzTopia made their first contribution in https://github.com/immich-app/immich/pull/7812
* @natedawg made their first contribution in https://github.com/immich-app/immich/pull/7910
* @growse made their first contribution in https://github.com/immich-app/immich/pull/7930
* @kokulshan made their first contribution in https://github.com/immich-app/immich/pull/7813
* @thielepaul made their first contribution in https://github.com/immich-app/immich/pull/7933
* @othyn made their first contribution in https://github.com/immich-app/immich/pull/7760
* @Aegeontis made their first contribution in https://github.com/immich-app/immich/pull/7851
* @sevtdy made their first contribution in https://github.com/immich-app/immich/pull/7942
* @dvdblg made their first contribution in https://github.com/immich-app/immich/pull/8018
* @tylerbrockett made their first contribution in https://github.com/immich-app/immich/pull/8017

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.98.2...v1.99.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.99.0'>v1.99.0</a>.</em>

**Comment by joelis10** [comment]
drag to select was one of my most missed [app who shall not be named] features, so happy it's here

**Reply by joelis10** [reply]
also local cache speeds up asset loading so much, might be the best QoL feature this app has had for lower powered servers

**Comment by OhaDerErste** [comment]
Dont know if this is intentional but new Icon is missing a Monochrome version

**Reply by joelis10** [reply]
can confirm missing for me too

**Reply by alextran1502** [reply]
Do you guys know how can enable monochrome mode to test generated monochrome icon?

**Reply by joelis10** [reply]
On Pixels: Settings > Wallpaper and Style > Themed Icons
Should be similar on other Android skins, might just say "Wallpaper"/"Style" instead of both

**Comment by Solarunit** [comment]
>OpenTelemetry integration
>This opt-in feature allows you to monitor Immich's performance. This data isn't sent anywhere beyond what you've configured.

Thanks for explanation, I was very surprised while reading this:

>There is a new endpoint for OpenTelemetry metrics to monitor performance.

Hope it always stays this way and not be enabled by default in the future.

**Reply by joelis10** [reply]
from what I understand from the docs (and I could be wrong) is that it requires a new container to be run so I don't think it'll ever be default

**Reply by bo0tzz** [reply]
This is all purely local. Data is not sent anywhere, it isn't possible for data to be sent anywhere unless you were to explicitly configure a separate container to do so, and we don't have any server for data to hypothetically be sent to. No worries :)

**Reply by palitu** [reply]
I wouldn't mind sending some stats to the team (anonymised of course).  things like features enabled, assets sync'd, device types, features used, etc.

In my mind, it is the least i could do to assist the project.  This sort of stuff helps them to validate any assumptions they make about what is and is not used by people

but of course - Opt-In and as anonymised as possible.

**Reply by jrasm91** [reply]
Yup, I think we may make it possible to send anonymous usage data in the future, but that isn't what this is.

**Comment by kaysond** [comment]
Very excited to try out the caching

**Reply by superkeyor** [reply]
waiting for updated ios version

**Comment by maicss** [comment]
"I would pay to see that", would you consider marking app is non-free, or 30 days free try out, to keep this project healthy running?

**Reply by alextran1502** [reply]
Hello, good question! We have no plan to make the app non-free. **The app will always be free with no pay-walled features.**

We might have an option for you to buy the app if you want, to replace the donation platforms we are using now. Part of it is to advocate to the users that you can and should pay for good open source software to support the development, this applies for the whole FOSS scene. And developers that build good software should not be afraid of asking for proper reward.

**Reply by Solarunit** [reply]
"keep this project healthy running" by making immich a non-free app? I personally would abandon such project.

**Reply by danieldietzler** [reply]
~~Sorry, maybe that wasn't really clear. I am pretty sure that what @alextran1502 was saying is that - for instance - we could provide hosted instances of Immich or move donations to a different platform. Nothing about this implies making Immich "non-free", as you mentioned. In fact, Alex stated in his very first sentences "We have *no* plan to make the app non-free. The app will *always* be free with no pay-walled features."~~ Nvm, you were probably referring to the original poster, right?

**Reply by maicss** [reply]
> "keep this project healthy running" by making immich a non-free app? I personally would abandon such project.

I mean iOS or android app non-free, not the main project, is this acceptable? I know maintainer said they wouldn't consider this option.

**Reply by jrasm91** [reply]
Yeah, that's not going to happen. If we were that desperate for money we'd do something else anyways.

**Reply by Solarunit** [reply]
@danieldietzler yup

**Comment by sadorowo** [comment]
when new version will be released on F-Droid?

**Reply by joelis10** [reply]
You can download the new version's APK directly from this release
